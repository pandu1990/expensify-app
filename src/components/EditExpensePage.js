import React from 'react';

const EditExpensePage = () => (
  <div>
    This is my edit expense page
  </div>
);

export default EditExpensePage;
