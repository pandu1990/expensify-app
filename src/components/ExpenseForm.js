import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      amount: props.expense ? (props.expense.amount / 100).toString() : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: undefined
    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };

  onCalendarFocusChange = ({ focused: calendarFocused }) => {
    this.setState(() => ({ calendarFocused }));
  };

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({ error: 'Please provide description and amount.' }));
    } else {
      this.setState(() => ({ error: undefined }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <div className="form__input-group">
          <label>Description</label>
          <input type="text" placeholder="Eg., Rent" autoFocus className="text-input" value={this.state.description} onChange={this.onDescriptionChange} />
        </div>
        <div className="form__input-group">
          <label>Amount</label>
          <input type="text" placeholder="1499.99" className="text-input" value={this.state.amount} inputMode="decimal" onChange={this.onAmountChange} />
        </div>
        <div className="form__input-group">
          <label>Date</label>
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onCalendarFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
        </div>
        <div className="form__input-group">
          <label>Note</label>
          <textarea placeholder="Add a note for your expense (optional)" className="textarea" value={this.state.note} onChange={this.onNoteChange}></textarea>
        </div>
        <div>
          <button className="button">Save Expense</button>
        </div>
      </form>
    );
  }
}
