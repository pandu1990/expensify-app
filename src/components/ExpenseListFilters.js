import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, setSortBy } from '../actions/filters';

const ExpenseListFilters = (props) => (
  <div>
    <input
      type="text"
      value={props.filters.text} onChange={(e) => {
        props.dispatch(setTextFilter(e.target.value));
      }}
    />
    <select
      value={props.filters.sortBy}
      onChange={(e) => {
        props.dispatch(setSortBy(e.target.value));
      }}
    >
      <option value="createdAt">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = (state) => ({
  filters: state.filters
});

export default connect(mapStateToProps)(ExpenseListFilters);
