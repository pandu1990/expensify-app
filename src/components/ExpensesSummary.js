import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div className="page-header">
    <div className="content-container">
      <h1 className="page-header__title">
        Viewing <span>{expenseCount}</span> {expenseCount === 1 ? 'expense' : 'expenses'} totalling <span>{numeral(expensesTotal / 100).format('$0,0.00')}</span>
      </h1>
      <div className="page-header__actions">
        <Link to="/create" className="button">Add Expense</Link>
      </div>
    </div>
  </div>
);

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: expenses.length,
    expensesTotal: selectExpensesTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
