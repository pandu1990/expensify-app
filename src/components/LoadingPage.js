import React from 'react';

const LoadingPage = () => (
  <div className="loader-container">
    <div className="loader"></div>
  </div>
);

export default LoadingPage;
