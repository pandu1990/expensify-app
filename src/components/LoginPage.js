import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startTwitterLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startTwitterLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It&apos;s time to get your expenses under control.</p>
      <button className="button button--login-google" onClick={startGoogleLogin} />
      <button className="button button--login-twitter" onClick={startTwitterLogin} />
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
  startTwitterLogin: () => dispatch(startTwitterLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
