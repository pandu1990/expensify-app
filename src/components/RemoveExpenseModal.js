import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = ({ showModal, closeModal, removeExpense }) => (
  <Modal ariaHideApp={false} isOpen={showModal} contentLabel="Remove Expense" onRequestClose={closeModal} closeTimeoutMS={200} className="modal">
    <h3 className="modal__title">Remove Expense</h3>
    <p className="modal__body">Are you sure you want to remove this expense?</p>
    <div className="modal__actions">
      <button
        className="button button--secondary button--modal"
        onClick={() => {
          removeExpense();
          closeModal();
        }}
      >
        Yes
      </button>
      <button className="button button--modal" onClick={closeModal}>
        No
      </button>
    </div>
  </Modal>
);

export default RemoveExpenseModal;
