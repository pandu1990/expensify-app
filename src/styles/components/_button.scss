.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--login {
  background: url('../images/btn_google_signin_dark.png');
  background-size: contain;
  width: 191px;
  height: 46px;
}

.button--link {
  background: none;
}

.button--secondary {
  background: $red;
}

.button--modal {
  padding: $s-size $l-size;
}
