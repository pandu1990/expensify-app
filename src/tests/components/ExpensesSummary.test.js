import React from 'react';
import { shallow } from 'enzyme';
import { test, expect } from '@jest/globals';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpenesesSummary correctly for one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={109500} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenesesSummary correctly for multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={234500} />);
  expect(wrapper).toMatchSnapshot();
});
