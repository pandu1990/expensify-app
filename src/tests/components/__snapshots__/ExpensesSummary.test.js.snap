// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenesesSummary correctly for multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $2,345.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenesesSummary correctly for one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1,095.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
